DSESwift/Sensor.swift

```
@@ -0,0 +1,109 @@
+        self.persistLocally = persistLocally
+        self.userId = userId
+        self.source = source
+        self.dataType = dataType
+        self.synced = synced
+    }
+    
+    public convenience init(name: String,   sensorOptions: SensorOptions,   userId: String,
+                            source: String, dataType: String, synced: Bool) {
+        self.init(
+            id: DatabaseHandler.getNextKeyForSensor(),
+            name: name,
+            meta: sensorOptions.meta,
+            csUploadEnabled: sensorOptions.uploadEnabled,
+            csDownloadEnabled: sensorOptions.downloadEnabled,
+            persistLocally: sensorOptions.persist,
+            userId: userId,
+            source: source,
+            dataType: dataType,
+            synced: synced
+        )
+    }
+    
+    convenience init(_ sensor: RLMSensor) {
+        self.init(
+            id: sensor.id,
+            name: sensor.name,
+            meta: sensor.meta,
+            csUploadEnabled: sensor.csUploadEnabled,
+            csDownloadEnabled: sensor.csDownloadEnabled,
+            persistLocally: sensor.persistLocally,
+            userId: sensor.userId,
+            source: sensor.source,
+            dataType: sensor.dataType,
+            synced: sensor.synced
+        )
+    }
+    /*
+    public func insertDataPoint(value: AnyObject, _date: NSDate) -> Bool {
+    
+        //Conversion from value to string
+        
+        let dataPoint = DataPoint(sensorId: DatabaseHandler.getNextKeyForSensor(), value: , date: date, synced: false)
+    }
+    */
+    
+    func getDataPoints(startDate: NSDate, endDate: NSDate, limit: Int, sortOrder: SortOrder) -> [DataPoint]{
+        var dataPoints = [DataPoint]()
+        do{
+            dataPoints = try DatabaseHandler.getDataPoints(sensorId: self.id, startDate: startDate, endDate: endDate, limit: limit, sortOrder: sortOrder)
+        } catch RLMError.InvalidLimit {
+            NSLog("Invalid limit was given.")
```

@kanekin why are the errors swallowed here instead of passed on?